module_utils/kafka_manager.py

```diff
@@ -895,31 +941,40 @@ def path = "/admin/reassign_partitions" ->
  and wait for its consumption if it is already present.
  Requires zk connection.
         """
+        topics_configuration = {}
+        for topic in topics:
```

Comment:
If I understand well, we are now generating 1 assignment for N topics (instead of N assignments for 1 topic), do you think it would be useful to have a `max_concurrent_assignment` options (or something similar) to avoid breaking a Kafka cluster if performing new assignments for many topics at the same time (maybe this is actually handled on Kafka side, I did not check)?

Reply:
Actually it's doing it per topic/partition (see https://github.com/apache/kafka/blob/a957748d2e0de872e881ec17f6cc27ada7ab7cd9/metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java#L1236). So if your doing it for all your topics/partitions it might take time but no more than specified on request https://github.com/apache/kafka/blob/a957748d2e0de872e881ec17f6cc27ada7ab7cd9/metadata/src/main/java/org/apache/kafka/controller/QuorumController.java#L1255 (might change our default value...)

Reply:
Added request timeout on all requests

Reply:
Thanks!